test/approvedTransfer.js

```diff
@@ -307,47 +307,86 @@ describe("Test Approved Transfer", function () {
+            let txReceipt = await manager.relay(transferModule, "approveTokenAndCallContract", [wallet.contractAddress, erc20.contractAddress, wallet.contractAddress, amountToApprove, wallet.contractAddress, dataToTransfer], wallet, [owner, ...sortWalletByAddress([guardian1, guardian2])]);
             const success = parseRelayReceipt(txReceipt);
-            assert.isNotOk(success, "approveTokenAndCall should fail wehn contract and wallet are the same");
+            assert.isNotOk(success, "approveTokenAndCall should fail when target contract is the wallet");
         });
 
-        it('should approve a token and call a contract with 1 EOA guardian and 2 smart-contract guardians', async () => {
+        it("should revert when target contract is an authorised module", async () => {
+            await addGuardians([guardian1, ...(await createSmartContractGuardians([guardian2, guardian3]))]);
+            let count = (await guardianManager.guardianCount(wallet.contractAddress)).toNumber();
+            assert.equal(count, 3, '3 guardians should be active');
+            
+            dataToTransfer = contract.contract.interface.functions['setStateAndPayToken'].encode([2, erc20.contractAddress, amountToApprove]);
+            let txReceipt = await manager.relay(transferModule, "approveTokenAndCallContract", [wallet.contractAddress, erc20.contractAddress, transferModule.contractAddress, amountToApprove, transferModule.contractAddress, dataToTransfer], wallet, [owner, ...sortWalletByAddress([guardian1, guardian2])]);
+            const success = parseRelayReceipt(txReceipt);
+            assert.isNotOk(success, "approveTokenAndCall should fail when target contract is an authorised module");
+        });
+
+        it('should approve token for a spender then call a contract with 3 guardians, spender = contract', async () => {
             await addGuardians([guardian1, ...(await createSmartContractGuardians([guardian2, guardian3]))]);
             let count = (await guardianManager.guardianCount(wallet.contractAddress)).toNumber();
             assert.equal(count, 3, '3 guardians should be active');
             let before = await erc20.balanceOf(contract.contractAddress);
             // should succeed with 2 confirmations
             dataToTransfer = contract.contract.interface.functions['setStateAndPayToken'].encode([2, erc20.contractAddress, amountToApprove]);
-            let txReceipt = await manager.relay(transferModule, "approveTokenAndCallContract", [wallet.contractAddress, erc20.contractAddress, contract.contractAddress, amountToApprove, dataToTransfer], wallet, [owner, ...sortWalletByAddress([guardian1, guardian2])]);
+            let txReceipt = await manager.relay(transferModule, "approveTokenAndCallContract", [wallet.contractAddress, erc20.contractAddress, contract.contractAddress, amountToApprove, contract.contractAddress, dataToTransfer], wallet, [owner, ...sortWalletByAddress([guardian1, guardian2])]);
```

Comment:
Add check the `txReceipt` is ok

Reply:
I've removed the `txReceipt` as it isn't used anywhere in the test and we already test for the functional success of the test.